runtime/doc/msgpack_rpc.txt

```diff
@@ -242,4 +242,150 @@ Even for statically compiled clients it is good practice to avoid hardcoding
+  notification with another {linedata} list, then these lines represent the
+  next N lines of the buffer. I.e., a second notification with another list of
+  100 strings will represent lines 101-200 of the buffer. If you send the
+  |nvim_buf_updates_start| request with its argument set to `"False"`, this
+  will be empty.
+
+  {linedata} will always have at least 1 item, but the maximum length is
+  determined by neovim and not guaranteed to be any particular size.  Also the
+  number of {linedata} items may vary between notifications, so your plugin
+  must be prepared to receive the line data in whatever size lists neovim
+  decides to split it into.
+
+  {more} is a boolean which tells you whether or not to expect more
+  |nvim_buf_updates_start| notifications. When {more} is false, you can be certain
+  that you now have the entire buffer's contents.
+
+nvim_buf_update[{buf}, {changedtick}, {firstline}, {lastline}, {linedata}] *nvim_buf_update*
+
+  Indicates that the lines between {firstline} and {lastline} (end-exclusive,
+  zero-indexed) have been replaced with the new line data contained in the
+  {linedata} list.  All buffer changes (even adding single characters) will be
+  transmitted as whole-line changes.
+
+	{buf} is an API handle for the buffer.
+
+  {changedtick} is the value of |b:changedtick| for the buffer. If you send an
+  API command back to neovim you can check the value of |b:changedtick| as
+  part of your request to ensure that no other changes have been made.
+
+  {firstline} is the integer line number of the first line that was replaced.
+  Note that {firstline} is zero-indexed, so if line `1` was replaced then
+  {firstline} will be `0` instead of `1`. {firstline} is guaranteed to always
+  be less than or equal to the number of lines that were in the buffer before
+  the lines were replaced.
+
+  {lastline} is the integer line number of the first line that was not replaced
+  (i.e. the range {firstline}, {lastline} is end-exclusive). Note that
+  {lastline} is zero-indexed, so if line numbers 2 to 5 were replaced, this
+  will be `5` instead of `6`. {lastline} is guaranteed to always be less than
+  or equal to the number of lines that were in the buffer before the lines were
+  replaced.
+
+  {linedata} is a list of strings containing the contents of the new buffer
+  lines. Newline characters are not included in the strings, so empty lines
+  will be given as empty strings. 
+
+  Note: sometimes {changedtick} will be |v:null|, which means that the buffer
+  text *looks* like it has changed, but actually hasn't. In this case the lines
+  in {linedata} contain the modified text that is shown to the user, but
+  doesn't reflect the actual buffer contents. Currently this behaviour is
+  only used for the 'inccommand' option.
+
+nvim_buf_update_tick[{buf}, {changedtick}]		                   *nvim_buf_update_tick*
+
+  Indicates that |b:changedtick| was incremented for the buffer {buf}, but no
```

Comment:
If this event is defined by `b:changedtick`, it should use the same jargon in the event name: `nvim_buf_changedtick` ?